distribution/integration/tests-integration/user-core/src/test/java/org/wso2/carbon/user/core/ldap/TestDirectoryServer.java

```diff
@@ -0,0 +1,117 @@
+package org.wso2.carbon.user.core.ldap;
+
+import org.apache.directory.server.core.DefaultDirectoryService;
+import org.apache.directory.server.core.DirectoryService;
+import org.apache.directory.server.core.entry.ServerEntry;
+import org.apache.directory.server.core.partition.Partition;
+import org.apache.directory.server.core.partition.impl.btree.jdbm.JdbmIndex;
+import org.apache.directory.server.core.partition.impl.btree.jdbm.JdbmPartition;
+import org.apache.directory.server.ldap.LdapServer;
+import org.apache.directory.server.protocol.shared.store.LdifFileLoader;
+import org.apache.directory.server.protocol.shared.transport.TcpTransport;
+import org.apache.directory.server.xdbm.Index;
+
+import java.io.File;
+import java.util.HashSet;
+
+/**
+ * Embedded directory server for testing.
+ */
+public class TestDirectoryServer {
+
+    /** The directory service */
+    private DirectoryService service;
+    private LdapServer ldapService;
+    private File workDir;
+
+    /**
+     * Main class. We just do a lookup on the server to check that it's available.
+     *
+     * @param args Not used.
+     */
+    public static void main(String[] args) //throws Exception
+    {
+        try {
+            // Create the server
+            TestDirectoryServer ads = new TestDirectoryServer();
+            ads.startLdapServer(12389);
+
+            Thread.sleep(100000);
+            ads.stopLdapService();
```

Comment:
This should go to finally block.

Reply:
No real value. Main program exists will anyway kills the in-memory LDAP server. Anyway, I will delete this method, as it is not used.